interface/usergroup/usergroup_admin_add.php

```diff
@@ -645,6 +655,52 @@ function authorized_clicked() {
 
 });
 </script>
+
+<!-- Password Strength Meter JavaScript -->
+<script>
+function checkPasswordStrength() {
+    console.log("checkPasswordStrength called");
+    var number = /([0-9])/;
+    var alphabets = /([a-zA-Z])/;
+    var special_characters = /([~,!,@,#,$,%,^,&,*,-,_,+,=,?,>,<])/;
+
+    var pwd = document.getElementById("stiltskin").value;
+    var strength = 0;
+
+    if (pwd.length < 6) {
+        document.getElementById('password_strength_meter').style.backgroundColor = "#ff6666";
+        document.getElementById('password_strength_text').innerHTML = "<?php echo xlt('Very Weak'); ?>";
+    } else {
+        if (pwd.match(number) && pwd.match(alphabets) && pwd.match(special_characters)) {
+            strength += 3;
+        } else if (pwd.match(number) && pwd.match(alphabets)) {
+            strength += 2;
+        } else if (pwd.match(alphabets)) {
+            strength += 1;
+        }
+
+        switch (strength) {
+            case 1:
+                document.getElementById('password_strength_meter').style.backgroundColor = "#ffcc00";
+                document.getElementById('password_strength_text').innerHTML = "<?php echo xlt('Weak'); ?>";
```

Comment:
Please use innerText calls instead of innerHTML.

Reply:
Resolved in my latest commit.